esp-wifi/src/wifi/mod.rs

```diff
@@ -198,6 +198,20 @@ pub enum Protocol {
     P802D11BGNAX,
 }
 
+/// A set of protocols.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Default)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub struct ProtocolSet(EnumSet<Protocol>);
+
+#[cfg(feature = "defmt")]
+impl defmt::Format for ProtocolSet {
+    fn format(&self, fmt: defmt::Formatter<'_>) {
+        for p in self.0 {
+            defmt::write!(fmt, "{}", p);
```

Comment:
How does the output look with this? I imagine concatenating protocols without a space inbetween might be a bit visually busy?